starter-code/routes/auth-routes.js

```diff
@@ -0,0 +1,194 @@
+// routes/auth-routes.js
+const express = require("express");
+const authRoutes = express.Router();
+const passport = require("passport");
+const ensureLogin = require("connect-ensure-login");
+
+
+// User model
+const User = require("../models/user");
+// Course model
+const Course = require("../models/course");
+
+// Bcrypt to encrypt passwords
+const bcrypt = require("bcrypt");
+const bcryptSalt = 10;
+
+// SIGN UP
+authRoutes.get("/signup", (req, res, next) => {
+  res.render("auth/signup");
+});
+
+authRoutes.post("/signup", (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+
+  if (username === "" || password === "") {
+    res.render("auth/signup", { message: "Indicate username and password" });
+    return;
+  }
+
+  User.findOne({ username }, "username", (err, user) => {
+    if (user !== null) {
+      res.render("auth/signup", { message: "The username already exists" });
+      return;
+    }
+
+    const salt = bcrypt.genSaltSync(bcryptSalt);
+    const hashPass = bcrypt.hashSync(password, salt);
+
+    const newUser = new User({
+      username,
+      password: hashPass
+    });
+
+    newUser.save((err) => {
+      if (err) {
+        res.render("auth/signup", { message: "Something went wrong" });
+      } else {
+        res.redirect("/login");
+      }
+    });
+  });
+});
+
+// CHECK ROLES FUNCTION
+
+function checkRoles(role) {
+  return function(req, res, next) {
+    if (req.isAuthenticated() && req.user.role === role) {
+      return next();
+    } else {
+      res.redirect('/login')
+    }
+  }
+}
+
+// LOG IN
+
+authRoutes.get("/login", (req, res, next) => {
+  res.render("auth/login", {"message": req.flash("error")});
+});
+
+authRoutes.post("/login", passport.authenticate("local", {
+  successRedirect: "/private-page",
+  failureRedirect: "/login",
+  failureFlash: true,
+  passReqToCallback: true
+}));
+
+// FB LOGIN
+
+authRoutes.get("/auth/facebook", passport.authenticate("facebook"));
+authRoutes.get("/auth/facebook/callback", passport.authenticate("facebook", {
+  successRedirect: "/private-page",
+  failureRedirect: "/login"
+}));
+
+// PRIVATE PAGES
+authRoutes.get("/private-page", (req, res, next) => {
+  next();
+},ensureLogin.ensureLoggedIn(), (req, res) => {
+  res.render("private", { user: req.user });
+});
+
+// ADD REMOVE USERS
+
+authRoutes.get("/addremove", checkRoles("Boss"),(req, res, next) => {
+  res.render("auth/addremove", {user: req.user});
+});
+
+authRoutes.post("/addremove", (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+  const role = req.body.role;
+  const name = req.body.name;
```

Comment:
Nice trick from ES6, these four  lines can be written like this:

const {username, password, role, name} = req.body

It will extract the props with these names and create variables with the same name by default.